src/gluonts/support/util.py

```diff
@@ -177,239 +177,245 @@ def erfinv(x: np.array) -> np.array:
+            for i, (x1, x2) in enumerate(zip(self.x_coord, self.x_coord[1:])):
+                if x1 < x < x2:
+                    denominator = x2 - x1 + self.tol
+                    return (x2 - x) / denominator * self.y_coord[i] + (
+                        x - x1
+                    ) / denominator * self.y_coord[i + 1]
 
 
 class TailApproximation:
     """
-    Apporoximate quantile function on tails based on trained quantile predictions
-    and make a inference on query point.
+    Approximate function on tails based on knots and make a inference on query point.
     Can be used for either interpolation or extrapolation on tails
 
     Parameters
     ----------
-    quantile_predictions
-        Dict with key = str(quantile) and value is np.array with size prediction_length
+    x_coord
+        x-coordinates of the data points must be in increasing order.
+    y_coord
+        y-coordinates of the data points - may be a higher numpy array.
     approximation_type
         str of tail approximation type
 
     """
 
     def __init__(
         self,
-        quantile_predictions: dict,
+        x_coord: List[float],
+        y_coord: List[np.ndarray],
         approximation_type: str = "exponential",
         tol: float = 1e-8,
-    ):
-        self.quantile_predictions = quantile_predictions
-        self.quantiles = sorted(map(float, self.quantile_predictions))
-        self.num_quantiles = len(self.quantile_predictions)
+    ) -> None:
+        self.x_coord = x_coord
+        assert sorted(self.x_coord) == self.x_coord
+        self.y_coord = y_coord
+        self.num_points = len(self.x_coord)
         self.approximation_type = (
-            approximation_type if self.num_quantiles > 1 else None
+            approximation_type if len(self.x_coord) > 1 else None
         )
         self.tol = tol
         if not self.approximation_type:
             pass
         elif self.approximation_type == "exponential":
-            self.init_exponential_tail_weights()
             self.tail_function_left = self.exponential_tail_left
             self.tail_function_right = self.exponential_tail_right
         else:
             raise NotImplementedError(
                 f"unknown approximation type {self.approximation_type}"
             )
 
-    def left(self, inference_quantile: float):
+    def left(self, x: float) -> np.ndarray:
         """
         Call left tail approximation
 
         Parameters
         -------
-        inference_quantile
-            float
+        x
+            x-coordinate to evaluate the left tail.
 
         Returns
         -------
-        same type and shape as any value of quantile_predictions dict,
-        i.e., type=np.array, shape = (prediction_length, )
+        np.ndarray
+            Interpolated values same shape as self.y_coord
         """
         if not self.approximation_type:
-            return self.quantile_predictions[str(self.quantiles[0])]
+            return self.y_coord[0]
```

Comment:
If I understand correctly, in this case we are saying that the function is constant everywhere to the left of the leftmost data point. This could be its own `ConstantTailApproximation` class implementing this

Reply:
yes, I think we can simplify the logic more and just remove this case since this would only happen in the case that there is only 1 point, which isn't supported.